indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java

```diff
@@ -176,6 +178,14 @@ public static NoopTask withGroupId(String groupId)
     return new NoopTask(null, groupId, null, 0, 0, null, null, null);
   }
 
+  public static NoopTask withJavaOptsContext(String javaOpts, List<String> javaOptsArray)
+  {
+    Map<String, Object> context = new HashMap<>();
```

Comment:
this should be `Map<String, Object> context = new HashMap<>(getContext());` so that we keep the original one as well